audio_service_platform_interface/lib/audio_service_platform_interface.dart

```diff
@@ -717,25 +752,24 @@ enum RatingStyleMessage {
 class OnPlaybackStateChangedRequest {
   final PlaybackStateMessage state;
 
-  OnPlaybackStateChangedRequest({
-    required this.state,
-  });
+  @literal
+  const OnPlaybackStateChangedRequest({required this.state});
 
   factory OnPlaybackStateChangedRequest.fromMap(Map map) =>
       OnPlaybackStateChangedRequest(
-          state: PlaybackStateMessage.fromMap(map['state']));
+        state: PlaybackStateMessage.fromMap(map['state']),
+      );
 
-  Map<dynamic, dynamic> toMap() => <dynamic, dynamic>{
```

Comment:
i removed `<dynamic, dynamic>` in all `toMap` methods, but in the next PR with lints i will add it back as <String, dynamic> since `implicit-dynamic: false` will be added
i don't see any strong reason to change it in terms of this PR

Reply:
I'm still not sure about this. Of course there is no problem with typing the map as `<String, dynamic>` before sending it through the platform channel except that that type information is lost immediately (the platform side won't use it). The problem I see is the reverse direction when receiving a map FROM the platform side. This will always be `<dynamic, dynamic>` and you can't just annotate `invokeMethod` with the type you want, that doesn't actually influence it's runtime type, only the static type. You will get a cast error at runtime when you try to do that, unless you use the `cast()` method. My choice to use `<dynamic, dynamic>` across platform channels is that it avoids needing to do this unnecessary `cast()` call which takes CPU cycles (not much, admittedly). So in order to justify this change to `<String, dynamic>` I would need to be convinced that this area of code is at risk to programmer error. But in reality the code is quite private and close to the wire of the method channel itself. I'm looking at the code that uses these map keys and I don't see the benefit in terms of safety.

Looking at plugins from Google, when dealing with method channels directly they also type their maps as `<dynamic, dynamic>` so to me that feels like confirmation that it's reasonable.

Reply:
> But in reality the code is quite private and close to the wire of the method channel itself. 

i want to emphasize that the main reason to do this if for a better inteface. you are definitely not the only one who uses/will use that, so keep it more understandable for anyone else - if you use `String`s as keys for maps, reflect this in the types

> Of course there is no problem with typing the map as <String, dynamic> before sending it through the platform channel except that that type information is lost immediately (the platform side won't use it).

this is unlikely, since it might be used for transfering data through channels. i'm not sure about that, but i wouldn't make any strong conclusions here

> You will get a cast error at runtime when you try to do that, unless you use the cast() method

you don't need `cast()` exlicitly because you can use `invokeMapMethod`, which will do this for you. if map has already a runtime type of string, the cast will make no effect. we should use the utilities language provides us with to build high quality interfaces. of course it will be a little bit harder than just snapping dynamics all over the place

> Looking at plugins from Google, when dealing with method channels directly they also type their maps as <dynamic, dynamic> so to me that feels like confirmation that it's reasonable.

this is not true, they mostly try to cast maps, and they almost do that if this is something public

they also tend to use `dynamic` in tests a lot, but i wouldn't say this is something good to follow since dynamic is always error-prone

what i would avoid (and what they do avoid in `flutter/plugins`) is using casts in `setMethodCallHandler`, becase this is really error-prone and unnecessary. i already fooled myself with this on branch with lints, because of `implicit-dynamic: false`. i will open an issue to discuss lints, synic this one in particular is very arguable for this plugin

Reply:
I didn't notice the `invokeMapMethod` and indeed you are using that. I checked more Google plugins and you're right that there are many instances of this usage, so I'm convinced.